gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/DataFetcherNode.java

```diff
@@ -37,6 +40,7 @@ public DataFetcherNode(
     this.limit = limit;
     this.offset = offset;
     this.orderByExpressionList = orderByExpressionList;
+    this.isPaginated = limit != null && offset != null;
```

Comment:
Given that the initial source of this is proto API, how do you know when it's not requested?

Reply:
when the creator of DataFetcherNode does not pass the limit and offset, it wouldn't be paginated

Reply:
ok.